src/ManagedArray.hpp

```diff
@@ -236,6 +248,12 @@ class ManagedArray {
 
   CHAI_HOST_DEVICE bool operator== (ManagedArray<T>& rhs);
 
+  template<bool B = std::is_base_of<CHAICopyable, T>::value, typename std::enable_if<B, int>::type = 0>
```

Comment:
Can you add comments clarifying when each implementation well be used.  

(I understand, but it would be helpful for future reference)

Reply:
Done.